appcell-resources/js/settings.js

```diff
@@ -1773,35 +1806,40 @@ st.passInputCheck = function(newpass, displayNameSpan) {
 }
 
 // API
-st.restCreateAccountAPI = function(json, pass) {
-  $.ajax({
-          type: "POST",
-          url: cm.user.cellUrl + '__ctl/Account',
-          data: JSON.stringify(json),
-          headers: {
-            'Authorization':'Bearer ' + cm.user.access_token,
-            'X-Personium-Credential': pass
-          }
-  }).done(function(data) {
-    if (document.getElementById("addCheckAccountLinkRole").checked) {
-      $("#ddlAddAccLinkRoleList option:selected").each(function(index, option) {
-        cm.setLinkParam($(option).text());
-        st.restAddAccountLinkRole(false);
-      });
+st.restCreateAccountAPI = function (json, pass) {
+    var headerObj = {};
+    headerObj.Authorization = 'Bearer ' + cm.user.access_token;
+    if (pass) {
+        headerObj["X-Personium-Credential"] = pass;
     }
-    st.getAccountList().done(function(data) {
-        st.dispAccountList(data);
+    $.ajax({
+        type: "POST",
+        url: cm.user.cellUrl + '__ctl/Account',
+        data: JSON.stringify(json),
+        headers: headerObj
+    }).done(function (data) {
+        if (!pass) {
+            cm.setLinkParam("admin([main])");
+            st.restAddAccountLinkRole(false);
+        }else if (document.getElementById("addCheckAccountLinkRole").checked) {
+            $("#ddlAddAccLinkRoleList option:selected").each(function (index, option) {
+                cm.setLinkParam($(option).text());
+                st.restAddAccountLinkRole(false);
+            });
+        }
+        st.getAccountList().done(function (data) {
+            st.dispAccountList(data);
+        });
```

Comment:
Please add fail callback to display error message for all newly implemented method.